src/parser/wat-parser.cpp

```diff
@@ -202,7 +202,10 @@ Result<> doParseModule(Module& wasm, Lexer& input, bool allowExtra) {
         CHECK_ERR(im);
       }
       if (!f->imported()) {
-        CHECK_ERR(ctx.irBuilder.visitEnd());
+        auto end = ctx.irBuilder.visitEnd();
+        if (auto* err = end.getErr()) {
+          return ctx.in.err(decls.funcDefs[i].pos, err->msg);
+        }
```

Comment:
Adds a source location to the error message. Otherwise you just get the plain error message where no indication of where the error occurred.

Reply:
I see, thanks. Is there a particular reason it is needed here and not elsewhere? (This location doesn't stand out to me.)

Reply:
In general the errors returned from parser methods already contain source locations because they were generated with `Lexer::err` in the first place. This error comes from `IRBuilder`, though, so it doesn't know anything about source locations and we have to add one.